src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/UDPCommunicator.java

```diff
@@ -0,0 +1,512 @@
+		if (this.isInitialized()) {
+			throw new IllegalStateException(ERROR_MESSAGE_CHANGE_PROPERTIES_AFTER_INIT);
+		} else {
+			this.bufferLength = bufferLength;
+		}
+	}
+
+	/**
+	 * This method returns the device UPD port
+	 *
+	 * @return int This returns the current UDP port.
+	 */
+	public int getPort() {
+		return this.port;
+	}
+
+	/**
+	 * This method is used set the device UDP port
+	 *
+	 * @param port This is the UDP port to set
+	 */
+	public void setPort(int port) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException(ERROR_MESSAGE_CHANGE_PROPERTIES_AFTER_INIT);
+		} else {
+			this.port = port;
+		}
+	}
+
+	/**
+	 * This method returns the login info
+	 *
+	 * @return String This return the current login info
+	 */
+	public String getLogin() {
+		return this.login;
+	}
+
+	/**
+	 * This method is used set the login info
+	 *
+	 * @param login This is current login info to set
+	 */
+	public void setLogin(String login) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException(ERROR_MESSAGE_CHANGE_PROPERTIES_AFTER_INIT);
+		} else {
+			this.login = login;
+		}
+	}
+
+	/**
+	 * This method returns the password info
+	 *
+	 * @return String This return the current password info
+	 */
+	public String getPassword() {
+		return this.password;
+	}
+
+	/**
+	 * This method is used set the password info
+	 *
+	 * @param password This is current password info to set
+	 */
+	public void setPassword(String password) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException(ERROR_MESSAGE_CHANGE_PROPERTIES_AFTER_INIT);
+		} else {
+			this.password = password;
+		}
+	}
+
+	/**
+	 * This method returns the list of command error
+	 *
+	 * @return List<String> This returns the current list of command error
+	 */
+	public List<String> getCommandErrorList() {
+		return this.commandErrorList;
+	}
+
+	/**
+	 * This method is used set the list of command error
+	 *
+	 * @param commandErrorList This is the list of command error to set
+	 */
+	protected void setCommandErrorList(List<String> commandErrorList) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException(ERROR_MESSAGE_CHANGE_PROPERTIES_AFTER_INIT);
+		} else {
+			this.commandErrorList = commandErrorList;
+		}
+	}
+
+	/**
+	 * This method returns the list of command success
+	 *
+	 * @return List<String> This returns the current list of command success
+	 */
+	public List<String> getCommandSuccessList() {
+		return this.commandSuccessList;
+	}
+
+	/**
+	 * This method is used set the list of command success
+	 *
+	 * @param commandSuccessList This is the list of command success to set
+	 */
+	protected void setCommandSuccessList(List<String> commandSuccessList) {
+		if (this.isInitialized()) {
+			throw new IllegalStateException(ERROR_MESSAGE_CHANGE_PROPERTIES_AFTER_INIT);
+		} else {
+			this.commandSuccessList = commandSuccessList;
+		}
+	}
+
+	/**
+	 * This method is used to create a connection
+	 */
```

Comment:
{@inheritDoc} on top and maybe more details on how it works below that :)

Reply:
thank you, I fixed it